bundled/tool/lsp_server.py

```diff
@@ -650,6 +650,27 @@ def get_cwd(settings: Dict[str, Any], document: Optional[workspace.Document]) ->
             return os.fspath(pathlib.Path(document.path).parent)
         return settings["workspaceFS"]
 
+    if settings["cwd"] == "${filePyproject}":
+        if document is None:
```

Comment:
we can probably move (`document is None`) case to the beginning of the function. Since it is common to all scenarios.